src/Symfony/Bundle/FrameworkBundle/Command/BuildDebugContainerTrait.php

```diff
@@ -32,13 +33,13 @@ trait BuildDebugContainerTrait
      *
      * @throws \LogicException
      */
-    protected function getContainerBuilder(): ContainerBuilder
+    protected function getContainerBuilder(KernelInterface $kernel = null): ContainerBuilder
     {
         if ($this->containerBuilder) {
             return $this->containerBuilder;
         }
 
-        $kernel = $this->getApplication()->getKernel();
+        $kernel = $kernel ?? $this->getApplication()->getKernel();
```

Comment:
Shouldn't it be a bit "better" if we make the kernel argument mandatory and remove `$this->getApplication()->getKernel()`? Also, we could move it to another namespace, as you noted it's not only useful for commands. And of course, we would have to update the current usage to pass the kernel instance.

This trait is internal so that we can improve it (no need to be done in this PR).

Reply:
Thank you. PR is submitted.

Reply:
I've rebased the PR.